### PR TITLE
pass weight_decay into optimizer

### LINKED ISSUE #191: weight_decay is not used in `finetuning.py`
### System Info

n/a

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

weight_decay is not used in `finetuning.py`
https://github.com/facebookresearch/llama-recipes/blob/c38bf5bdd370ceb93e71cfec1a07b0885a57e3ec/src/llama_recipes/finetuning.py#L222-L235 even though it's part of the config https://github.com/facebookresearch/llama-recipes/blob/c38bf5bdd370ceb93e71cfec1a07b0885a57e3ec/src/llama_recipes/configs/training.py#L18

### Error logs

n/a

### Expected behavior

`weight_decay` should be passed into the optimizer. on a related note, since it's hard-coded to 0, is it recommended to set `weight_decay` to 0?

### PR DESCRIPTION
**Pull request by shijie-wu**
# What does this PR do?

@mreso Fixes #191

## Feature/Issue validation/testing

This is a bug fix on actually using weight decay and should not impact any existing workload.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/llama-recipes/blob/main/CONTRIBUTING.md#pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?  
- [ ] Did you write any new necessary tests?

Thanks for contributing 🎉!

### CODE REVIEW

**Review by mreso — APPROVED** [review]
Thanks for the PR! Added a unit test
